ultimatepython/data_structures/heap.py

```diff
@@ -0,0 +1,66 @@
+"""
+Heaps are a special type of binary tree-based data structure that
+maintains a specific order among the elements. This tutorial covers
+how to create, manipulate, and use heaps in Python.
+"""
+
+import heapq
+
+
+def main():
+    # Creating a min-heap
+    min_heap = []
+    heapq.heappush(min_heap, 3)
+    heapq.heappush(min_heap, 1)
+    heapq.heappush(min_heap, 4)
+    heapq.heappush(min_heap, 1)
+    heapq.heappush(min_heap, 5)
+    assert min_heap[0] == 1  # The root of the heap is the smallest element
+
+    # Pop the smallest element

```

Comment:
Should we also make an assertion about `min_heap[0]` around this area as well? It might be good for readers to know that the min heap variant continues to stay the same throughout the operations